core/src/main/java/hudson/model/Queue.java

```diff
@@ -1522,32 +1535,31 @@ public void maintain() {
     private @CheckForNull Runnable makeBuildable(final BuildableItem p) {
         if (p.task instanceof FlyweightTask) {
             if (!isBlockedByShutdown(p.task)) {
-            Jenkins h = Jenkins.getInstance();
-            Map<Node,Integer> hashSource = new HashMap<Node, Integer>(h.getNodes().size());
+                Jenkins h = Jenkins.getInstance();
+                Map<Node,Integer> hashSource = new HashMap<Node, Integer>(h.getNodes().size());
 
-            // Even if master is configured with zero executors, we may need to run a flyweight task like MatrixProject on it.
-            hashSource.put(h, Math.max(h.getNumExecutors() * 100, 1));
+                // Even if master is configured with zero executors, we may need to run a flyweight task like MatrixProject on it.
+                hashSource.put(h, Math.max(h.getNumExecutors() * 100, 1));
 
-            for (Node n : h.getNodes()) {
-                hashSource.put(n, n.getNumExecutors() * 100);
-            }
+                for (Node n : h.getNodes()) {
+                    hashSource.put(n, n.getNumExecutors() * 100);
+                }
 
-            ConsistentHash<Node> hash = new ConsistentHash<Node>(NODE_HASH);
-            hash.addAll(hashSource);
-
-            Label lbl = p.getAssignedLabel();
-            for (Node n : hash.list(p.task.getFullDisplayName())) {
-                final Computer c = n.toComputer();
-                if (c==null || c.isOffline())    continue;
-                if (lbl!=null && !lbl.contains(n))  continue;
```

Comment:
Is there a reason this check was deleted here? (Much more visible in the [whitespace-ignoring diff](https://github.com/jenkinsci/jenkins/pull/1815/files?w=1).)